### PR TITLE
feat(packager): add noInstall option

### LINKED ISSUE #913: Why does serverless-webpack run yarn install for you?
# This is a Question

## Description

See: https://github.com/serverless-heaven/serverless-webpack/blob/master/lib/packagers/yarn.js

serverless-webpack currently runs yarn install (as well as npm install) for you, depending on packager config. However, I can't help but feel that this is a bit redundant. Wouldn't most CI workflows already have run install prior to running serverless webpack? If not, how would serverless webpack even be installed to be able to run the command itself?

This is an issue particularly because it breaks yarn PnP. Are you open to at least having a config to disable the install altogether?

### PR DESCRIPTION
**Pull request by russell-dot-js**
## What did you implement:

Closes #913

## How did you implement it:

* Add noInstall packagerOption for both npm and yarn
* Update the install methods to skip install if noInstall is true
* Update tests to demonstrate

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** YES

### CODE REVIEW

**Comment by vicary** [comment]
I can see the implementation idea comes from the current `--no-build` option, but since 2.0 Serverless CLI warns about out of spec CLI options. Unfortunately `--no-build` and the new `--no-install` is among one of them.

I would suggest supporting a new option in `serverless.yml` which takes precedence of the CLI option. See how `stage` works in the [official docs](https://www.serverless.com/framework/docs/providers/aws/guide/variables/) for an idea.

To accommodate both usage, how about reading the CLI option value only when the new option in `serverless.yml` is not set? For users who wants to use CLI option as a fallback, we can recommend the use the of syntax `${opt:install, ...}` in README.md.

**Review by vicary — APPROVED** [review]
This could be a minor release.

File: README.md [hunk 1]
```diff
@@ -371,7 +371,11 @@ configuration setting. For details see below.
 By default, the plugin uses NPM to package the external modules. However, if you use npm,
 you should use any version `<5.5 >=5.7.1` as the versions in-between have some nasty bugs.
 
-Right now there are no `packagerOptions` that can be set with NPM.
+The NPM packager supports the following `packagerOptions`:
+
+| Option             | Type | Default | Description                                         |
+| ------------------ | ---- | ------- | --------------------------------------------------- |
+| noInstall          | bool | false   | Do not run npm install (assume install completed)   |
```
Comment (j0k3r) [inline README.md:378]:
```suggestion
| noInstall          | bool | false   | Do not run `npm install` (assume install completed)   |
```

File: README.md [hunk 2]
```diff
@@ -382,6 +386,7 @@ The yarn packager supports the following `packagerOptions`:
 | Option             | Type | Default | Description                                         |
 | ------------------ | ---- | ------- | --------------------------------------------------- |
 | ignoreScripts      | bool | false   | Do not execute package.json hook scripts on install |
+| noInstall          | bool | false   | Do not run yarn install (assume install completed)  |
```
Comment (j0k3r) [inline README.md:389]:
```suggestion
| noInstall          | bool | false   | Do not run `yarn install` (assume install completed)  |
```

File: README.md [hunk 3]
```diff
@@ -502,6 +507,7 @@ custom:
 This can be useful, in case you want to upload the source maps to your Error
 reporting system, or just have it available for some post processing.
 
+
```
Comment (j0k3r) [inline README.md:510]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -515,7 +521,6 @@ exampleFunction:
 
 ⚠️ **Note: this will only work if your custom runtime and function are written in JavaScript.
 Make sure you know what you are doing when this option is set to `true`**
-
```
Comment (j0k3r) [inline README.md:518]: To revert

**Review by j0k3r — COMMENTED** [review]
Just some small typos

**Pull request closed without merging**